src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -186,7 +190,7 @@ public Set<String> runBinaryScanner(Set<String> currentFeatureSet, List<String>
                 } else if (scannerException.getClass().getName().equals(FEATURE_MODIFIED_EXCEPTION)) {
                     // The scanned files conflict and the scanner suggests modifying some features
                     Set<String> modifications = getFeatures(scannerException);
-                    Set<String> sampleFeatureList = reRunBinaryScanner(allClassesDirectories, logLocation, targetJavaEE, targetMicroProfile);
+                    Set<String> sampleFeatureList = reRunIfFailed ? reRunBinaryScanner(allClassesDirectories, logLocation, targetJavaEE, targetMicroProfile) : null;
```

Comment:
For the `FEATURE_MODIFIED_EXCEPTION` case, we should use the `featureList` from the first call as the `sampleFeatureList` if we are not making a second call, as we will want this list either recommended or applied depending on the scenario.

Reply:
oh I see, I thought the `sampleFeatureList` was used for the recommendation/apply logic. Since it is not, I agree, we should use null.